### ISSUE TITLE
Does netlify dev --live support redirect shadowing?

### ISSUE THREAD

**Issue by pixelprogrammer** [netlify/cli#920]
We have a rule in the redirects that catches pages that don't exist and redirects them to a 404 page. However, it seems to be catching all pages and static assets that do exist and redirecting. This same redirect file works when live but when using the `netlify dev --live` command the redirect rules don't work.

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**- Do you want to request a _feature_ or report a _bug_?**

BUG

**- What is the current behavior?**

Visiting static asset urls that exist redirect because the redirect rule catches all

**- If the current behavior is a bug, please provide the steps to reproduce.**

Edit your `_redirects` file to include this at the end:

```
/* /404
```

Visit any page and all your static assets should be left in some kind of 301 redirect loop.

**- What is the expected behavior?**

Pages and assets that exist are not redirected and follow shadow pattern that live sites respect: [Netlify Shadowing](https://docs.netlify.com/routing/redirects/rewrites-proxies/#shadowing)

**- Local Environment Information**
```
────────────────────┐
 Environment Info   │
────────────────────┘

System:
    OS: Linux 5.6 Fedora 32 (Workstation Edition) 32 (Workstation Edition)
    CPU: (4) x64 Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
  Binaries:
    Node: 12.16.3 - /usr/bin/node
    Yarn: 1.21.1 - /usr/bin/yarn
    npm: 6.14.4 - /usr/bin/npm
  Browsers:
    Chrome: 81.0.4044.138
    Firefox: 76.0.1
  npmGlobalPackages:
    netlify-cli: 2.51.0

netlify-cli/2.51.0 linux-x64 node-v12.16.3

```

**Comment by RaeesBhatti** [comment]
@pixelprogrammer Thank you for opening this issue. `netlify dev` looks for static files in your `publish` directory, and falls back to redirect if no file exists.
Can you please share the `build` and `dev` sections of your `netlify.toml` and your project's directory structure!

**Comment by pixelprogrammer** [comment]
Sure thing:

**netlify.toml**

```toml
[build]
  command = "npm run build"
  publish = "dist"

[dev]
  framework = "nuxt"
  command = "npm run dev"
  publish = "dist"
  autoLaunch = true
```

**Directory Structure**
```
dist/
  _nuxt/
  _redirects
  some-image.png
  ...etc
static/
  _redirects
  some-image.png
  ...etc
components/
pages/
netlify.toml
nuxt.config.js
...etc
  
```

So in that example, if I go to `https://website.com/some-image.png` I get a redirect loop for that image. Same with assets within the `_nuxt` folder.

**Comment by overlordofmu** [comment]
Hi @erezrokah . I'm able reproduce this failure to shadow files with the current version of `netlify-cli`:

```text
$ netlify version
netlify-cli/2.64.1 darwin-x64 node-v10.16.0
$ tree
.
├── README.md
├── base
│   ├── copy-build.sh
│   ├── netlify.toml
│   ├── post
│   │   ├── Page-2
│   │   │   └── index.html
│   │   ├── green-river.png
│   │   ├── index.html
│   │   └── large.jpg
│   └── pre
│       ├── Page-2
│       │   └── index.html
│       ├── _redirects
│       ├── ads.txt
│       ├── green-river.png
│       ├── index.html
│       ├── large.jpg
│       ├── logo.svg
│       └── test.svg
├── exiter.sh
└── netlify.toml
```
 
If `netlify.toml` contains this:

```text
[[redirects]]
  from = "/*"
  to = "/en/:splat"
  force = false
```

The following happens:

```text
$ netlify dev -d base/post
◈ Netlify Dev ◈
◈ Injected addon env var:  NETLIFY_LFS_ORIGIN_URL
◈ Injected build setting env var: CI
◈ Using simple static server because --dir flag was specified
◈ Running static server from "base/post"

◈ Server listening to 3999

   ┌─────────────────────────────────────────────────┐
   │                                                 │
   │   ◈ Server now ready on http://localhost:8888   │
   │                                                 │
   └─────────────────────────────────────────────────┘
```

In another terminal I see this:

```text
$ curl -svo /dev/null http://localhost:8888/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8888 (#0)
> GET / HTTP/1.1
> Host: localhost:8888
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Location: /en/
< Cache-Control: no-cache
< Date: Wed, 30 Sep 2020 08:50:45 GMT
< Connection: keep-alive
< Transfer-Encoding: chunked
<
{ [30 bytes data]
* Connection #0 to host localhost left intact
* Closing connection 0
```

However, for the same setup, if I remove the redirect, I get the 200 response:

```text
$ curl -svo /dev/null http://localhost:8888/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8888 (#0)
> GET / HTTP/1.1
> Host: localhost:8888
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< x-powered-by: netlify-dev
< date: Wed, 30 Sep 2020 08:57:24 GMT
< content-type: text/html
< content-length: 437
< connection: close
<
{ [437 bytes data]
* Closing connection 0
```

The only change for the second `curl` request is that the redirect rule was deleted from `netlify.toml`. As the rule isn't forced, it should always be a 200 response even with the redirect.

In other words, the file shadowing doesn't appear to be working and the redirect rule is always used whether the file exists or not. I can make the demo repo I used for this public if you like.

**Comment by erezrokah** [comment]
I believe this was fixed by https://github.com/netlify/cli/pull/2195, so I'm closing the issue.

If it still reproduces in the latest version, please comment and I'll re-open
